src/proto/streams/send.rs

```diff
@@ -335,7 +339,10 @@ impl Send {
         if available as usize <= buffered {
             0
         } else {
-            available - buffered as WindowSize
+            cmp::min(
+                available - buffered as WindowSize,
+                self.prioritize.available(),
+            )
```

Comment:
The bug was using `self.prioritize.available()` here.

The capacity assigned to any given stream was already capped by the currently available buffer size, and the connection-level capacity was decreased accordingly, so we end up applying a min with 0.